File: trunk/quests/potactics/Vallon_Zek.pl
sub EVENT_SPAWN {
	# one second delay so everything processes
	quest::settimer("vzspawned",1);
}

sub EVENT_TIMER {
	if ($timer eq "vzspawned") {
		# tell vallon_controller i spawned
		quest::stoptimer("vzspawned");
		quest::signalwith(214112,1);
	}
}

sub EVENT_SIGNAL {
	#signal value equals which phase we are on
	if ($signal == 1) {
		#phase 1, depop at 75%
		quest::setnexthpevent(76);
	} elsif ($signal == 2 || $signal == 3) {
		#phase 2 and 3, depop at 50%
		quest::setnexthpevent(51);
	}
}

sub EVENT_HP {
	# tell vallon_controller i have depopped.
	quest::signalwith(214112,2);
	quest::depop();
}

sub EVENT_DEATH_COMPLETE {
	# spawn the planar projection on my corpse
	quest::spawn2(202368,0,0,$x,$y,$z,$h);
	# tell vallon_controller I have died
	quest::signalwith(214112,3);
}

File: potactics/Vallon_Zek.lua
function event_spawn(e)
	eq.set_timer("vzspawned",1000);
	--one second delay so everything processes
end

function event_timer(e)
	if (e.timer == "vzspawned") then
		--tell vallon_controller I spawned
		eq.stop_timer("vzspawned");
		eq.signal(214112,1); -- NPC: #vallon_controller
	end
end

function event_signal(e)
	--signal value equals which phase we are on
	if (e.signal == 1) then
		--phase 1, depop at 75%
		eq.set_next_hp_event(76);
	elseif (e.signal == 2 or e.signal == 3) then
		--phase 2 and 3 depop at 50%
		eq.set_next_hp_event(51);
	end
end

function event_hp(e)
	--tell vallon_controller I have depopped
	eq.signal(214112,2); -- NPC: #vallon_controller
	eq.depop();
end

function event_death_complete(e)
	--spawn planar projection
	local xloc = e.self:GetX();
	local yloc = e.self:GetY();
	local zloc = e.self:GetZ();
	eq.spawn2(202368,0,0,xloc,yloc,zloc,0); -- NPC: A_Planar_Projection
	--tell vallon_controller I died
	eq.signal(214112,3); -- NPC: #vallon_controller
end